src/main/java/com/kainos/ea/controllers/RoleController.java

```diff
@@ -43,4 +42,29 @@ public Response getAllJobRoles() throws SQLException {
                     .build();
         }
     }
+    @GET
+    @Path("/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed({UserRole.ADMIN_USER, UserRole.APPLICANT_USER})
+    @ApiOperation(
+            value = "Returns a Job Role by ID",
+            authorizations = @Authorization(value = HttpHeaders.AUTHORIZATION),
+            response = List.class)
+    public Response getRoleById(@PathParam("id") final int id) {
+        try {
+            return Response.ok().entity(roleService.getRoleById(id)).build();
+        } catch (FailedToRetrieveException | JobRoleNotFoundException e) {
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity("An error occurred while retrieving "
```

Comment:
Oh I see, perhaps these error messages should be different? We've caught separate errors intentionally but then we're sending back the same error message